packages/flutter/lib/src/cupertino/sheet_route.dart

```diff
@@ -0,0 +1,254 @@
+            return _CupertinoSheetDecorationBuilder(
+              child: Builder(
+                builder: (BuildContext context) {
+                  return builder(context);
+                },
+              ),
+              backgroundColor: backgroundColor,
+              topRadius: _kCupertinoSheetTopRadius,
+            );
+          },
+          settings: settings,
+          animationCurve: _kCupertinoSheetCurve,
+          stops: stops,
+          initialStop: initialStop,
+        );
+
+  @override
+  bool get bounceAtTop => true;
+
+  @override
+  bool get draggable => true;
+
+  @override
+  bool get expanded => true;
+
+  @override
+  Color? get barrierColor => _kBarrierColor;
+
+  @override
+  bool get barrierDismissible => false;
+
+  @override
+  Widget buildTransitions(
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    final double topPadding = MediaQuery.of(context)!.padding.top;
+    final double topOffset = math.max(_kSheetMinimalOffset, topPadding);
+    return AnimatedBuilder(
+      animation: secondaryAnimation,
+      child: child,
+      builder: (BuildContext context, Widget? child) {
+        final double progress = secondaryAnimation.value;
+        final double scale = 1 - progress / 10;
+        final double distanceWithScale = (topOffset + _kPreviousRouteVisibleOffset) * 0.9;
+        final Offset offset = Offset(0, progress * (topOffset - distanceWithScale));
+        return Transform.translate(
+          offset: offset,
+          child: Transform.scale(
+            scale: scale,
+            child: child,
+            alignment: Alignment.topCenter,
+          ),
+        );
+      },
+    );
+  }
+
+  @override
+  Widget getBottomRouteTransition(
+      BuildContext context, Animation<double> secondAnimation, Widget child) {
+    final Animation<double> delayAnimation = initialStop == 1
+        ? secondAnimation
+        : CurvedAnimation(
+            parent: sheetAnimation!,
+            curve: const Interval(0.5, 1, curve: Curves.linear),
+          );
+
+    return _CupertinoSheetBottomRouteTransition(
+      body: child,
+      secondaryAnimation: delayAnimation,
+    );
+  }
+}
+
+/// Animation for previous route when a [CupertinoSheetRoute] enters/exits
+class _CupertinoSheetBottomRouteTransition extends StatelessWidget {
+  
+  const _CupertinoSheetBottomRouteTransition({
+    Key? key,
+    required this.secondaryAnimation,
+    required this.body,
+  }) : super(key: key);
+
+  final Widget body;
+
+  final Animation<double> secondaryAnimation;
+
+  @override
+  Widget build(BuildContext context) {
+    final double topPadding = MediaQuery.of(context)!.padding.top;
+    final double topOffset = math.max(_kSheetMinimalOffset, topPadding);
+
+    final bool isRoundedDevice =
+        defaultTargetPlatform == TargetPlatform.iOS && topPadding > _kRoundedDeviceStatusBarHeight;
```

Comment:
is there a more reliable way of determining if it's a "rounded device"? e.g. is the recent MacBook Pro considered "rounded"? (I guess not because the roundness is where the menu bar goes?)